### PR TITLE
Add info about viewing compiled docs via www at NERSC

### PR DESCRIPTION
**Pull request by matthewhoffman**
This merge adds information about how to view compiled documentation via NERSC's web Science Gateway to avoid needing to scp the compiled documentation locally.

Checklist

* [x] Developer's Guide has been updated
* [x] Documentation has been [built locally](https://mpas-dev.github.io/compass/latest/developers_guide/building_docs.html) and changes look as expected

### CODE REVIEW

**Comment by matthewhoffman** [comment]
Additional information may need to be added about setting the permissions correctly on files and dirs in the www directory so they are visible from the internet.  (cc: @andrewdnolan )

**Review by xylar — APPROVED** [review]
Yep, that looks good.  We could add similar instructions for Compy, Anvil and Chrysalis in the future since they all have web portals.

**Comment by xylar** [comment]
@matthewhoffman, maybe we can get this finished up (see suggestion above).

File: docs/developers_guide/building_docs.rst [hunk 1]
```diff
@@ -23,3 +23,17 @@ script was last sourced.  The load script will reinstall compass into the conda
...
+On NERSC, you can conveniently make use of their
+`web Science Gateway <https://docs.nersc.gov/services/science-gateways/#web-methods-for-data>`_
+to avoid transferring data.  By copying your compiled documentation to
+an appropriately named subdirectory at
+``/global/cfs/cdirs/<myproj>/www/`` and then running ``chmod -R ugo+rX .`` on that directory 
+(and perhaps its parent directories), it can then be viewed from a browser at
+``https://portal.nersc.gov/cfs/<myproj>/``.
+
+LCRC also has a web portal.  You can copy the compiled documentation to a subdirectory in 
+``/lcrc/group/e3sm/public_html/diagnostic_output/<username>/``.  Then, you can view you can
```
Comment (matthewhoffman) [inline docs/developers_guide/building_docs.rst:38]:
```suggestion
``/lcrc/group/e3sm/public_html/diagnostic_output/<username>/``.  Then, you can
```

**Comment by matthewhoffman** [comment]
@xylar , thanks for providing the suggestion for this.  I committed a modified version of your suggestion and confirmed the instructions worked correctly on Anvil and the results look correct.  I have not tried the updated instructions with the chmod command for Perlmutter, but I can try that tomorrow if it's back up if you haven't confirmed it's correct already.

**Comment by xylar** [comment]
@matthewhoffman, yes, the command I added is one I use on Perlmutter all the time so I know that it is correct.

**Pull request merged**